doc/api/cli.md

```diff
@@ -1016,6 +1016,13 @@ variables which configure Node.js][environment_variables], such as `NODE_OPTIONS
 are parsed and applied. If the same variable is defined in the environment and
 in the file, the value from the environment takes precedence.
 
+You can pass multiple `--env-file` arguments. Subsequent files override
+pre-existing variables defined in previous files.
```

Comment:
@anonrig I find this behavior confusing, as it is "the opposite" then with envs that pre-exist on the system.

Since we preserve values from system and do not override them, I would expect that reading the files would act similarly, and any file we read will not override any value previously defined.

WDYT?

Reply:
Ideally there should be a flag to control this behaviour. Current behaviour is what most users need, but not what many libraries do by default.

Reply:
We should be somewhat careful not to add too much complexity to our already vastly overloaded CLI.

Reply:
@atlowChemi If we go with not overriding, then first `--env-file` will define process.env.BASIC and the rest can't update it. Right now, The second `--env-file` will update process.env.BASIC (if it's defined by the previous declaration, not as a process.env variable). 

I think this approach is better than not updating it.

Reply:
>  If we go with not overriding, then first `--env-file` will define process.env.BASIC and the rest can't update it. Right now, The second `--env-file` will update process.env.BASIC (if it's defined by the previous declaration, not as a process.env variable).

I understand that 🙂 
I find this behavior inconsistent, as if it were defined on the environment (which is "higher hierarchy" then any `--env-file`), no file would override it, where as within `--env-file`s the "lower hierarchy" is the the "stronger" value.

I don't feel to strongly about this, and I truly appreciate the work you put in to this awesome feature, I felt this is an inconsistency worth noting and discussing 🙂

Reply:
Thank you @atlowChemi! I appreciate it.

I understand your concern, but I feel like we are comparing apples with oranges. If you feel strongly about this change, you have the right to block this, and we can discuss it more deeply. No harm done :)

IMHO: I feel this is the correct path to move forward